Android/Samples/libraries/kml/src/test/java/com/microsoft/maps/kml/KMLParserTest.java

```diff
@@ -3,15 +3,103 @@
 
 package com.microsoft.maps.kml;
 
+import com.microsoft.maps.MockBingMapsLoader;
+import com.microsoft.maps.moduletools.MapFactories;
+import com.microsoft.maps.moduletoolstest.MockParserMapFactories;
+import java.io.IOException;
+import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+import org.xmlpull.v1.XmlPullParserException;
 
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = Config.NONE)
 public class KMLParserTest {
 
+  private static final MapFactories MOCK_MAP_FACTORIES = new MockParserMapFactories();
+
+  @Before
+  public void setup() {
```

Comment:
Add some tests with xml comments in various places to ensure it's handled properly.